src/modelfree/score_agent.py

```diff
@@ -105,18 +154,40 @@ def env_fn(i):
     if record_traj:
         venv.save(save_dir=record_traj_params['save_dir'])
 
+    if videos:
+        for env_video_dir in video_dirs:
+            try:
+                for video_file_path in os.listdir(env_video_dir):
+                    if 'mp4' in video_file_path:
+                        env_number = env_video_dir.split("/")[-1]
+                        sacred_name = "env_{}_{}".format(env_number, video_file_path)
+                        score_ex.add_artifact(filename=os.path.join(env_video_dir,
+                                                                    video_file_path),
+                                              name=sacred_name)
+            except FileNotFoundError:
+                warnings.warn("Can't find path {}; no videos from that path added as artifacts"
+                              .format(env_video_dir))
+
+        if tmp_dir is not None:
+            tmp_dir.cleanup()
+
+    for observer in score_ex.observers:
+        if hasattr(observer, 'dir'):
+            _clean_video_directory_structure(observer)
+
     for agent in agents:
         if agent.sess is not None:
             agent.sess.close()
-    venv.close()
 
+    venv.close()
     return score
 
 
 def main():
     observer = FileStorageObserver.create(osp.join('data', 'sacred', 'score'))
     score_ex.observers.append(observer)
     score_ex.run_commandline()
+    logging.info("Sacred run completed, files stored at {}".format(observer.dir))
```

Comment:
I tend to favour having `logger = logging.getLogger('module.name')` at the top level, and then call `logger` rather than `logging`. This lets you have per-module logging levels (e.g. debug in one module, info everywhere else) and will include the module name in logging output. Doesn't matter too much when it's a CLI script, but really useful for library code.

Reply:
I initially decided not to do that because it didn't seem like there was a clean way to have the same logger object visible in both the main() method and in the sacred experiment, unless we defined it globally, which I've done now.